File: RecipeBook/RecipeBook/Models/Domain/Recipe.cs
namespace RecipeBook.Models.Domain
{
    public class Recipe
    {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public int Steps { get; set; }
        public int Portions { get; set; }
        public int Duration { get; set; }
        public Enums.DomainEnums.Difficulty Difficulty { get; set; }

        //Navigation Property
        public IEnumerable<Ingredient> Ingredients { get; set; }
    }
}

File: RecipeBook/client-app/src/app/models/recipe.ts
export interface Recipe {
    id: string;
    title: string;
    description: string;
    steps: number;
    portions: number;
    duration: number;
    difficulty: Difficulty;
    imageName: string;
    ingredients: Ingredient[];
}

export interface Ingredient {
    id: string;
    ingredientName: IngredientName;
    ingredientQuantity: IngredientQuantity;
}

export interface IngredientName {
    id: string;
    name: string;
}

export interface IngredientQuantity {
    id: string;
    quantity: string;
}

export enum Difficulty {
    VeryEasy = 0,
    Easy = 1,
    Medium = 2,
    Hard = 3,
    VeryHard = 4
}
